### ISSUE TITLE
RuntimeError: quantile() input tensor must be either float or double dtype

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#103054]
### 🐛 Describe the bug

```
│ /data/users/ezyang/a/pytorch/torch/_inductor/utils.py:63 in do_bench                             │
│                                                                                                  │
│     60 │                                                                                         │
│     61 │   if quantile_field_name not in kwargs:                                                 │
│     62 │   │   kwargs[quantile_field_name] = (0.5, 0.2, 0.8)                                     │
│ ❱   63 │   return triton_do_bench(*args, **kwargs)[0]                                            │
│     64                                                                                           │
│     65                                                                                           │
│     66 @functools.lru_cache(None)                                                                │
│                                                                                                  │
│ /home/ezyang/local/a/pytorch-env/lib/python3.10/site-packages/triton/testing.py:87 in do_bench   │
│                                                                                                  │
│    84 │   torch.cuda.synchronize()                                                               │
│    85 │   times = torch.tensor([s.elapsed_time(e) for s, e in zip(start_event, end_event)])      │
│    86 │   if quantiles is not None:                                                              │
│ ❱  87 │   │   ret = torch.quantile(times, torch.tensor(quantiles)).tolist()                      │
│    88 │   │   if len(ret) == 1:                                                                  │
│    89 │   │   │   ret = ret[0]                                                                   │
│    90 │   │   return ret                                                                         │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
BackendCompilerFailed: backend='inductor' raised:
RuntimeError: quantile() input tensor must be either float or double dtype
```

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov

**Comment by ezyang** [comment]
https://github.com/openai/triton/pull/1741

**Comment by ego-thales** [comment]
Since it was closed as "completed", might I ask if there is any good reason to restrict `quantile` to `float`/`double`? I find this kind of weird and very restrictive.

**Comment by ProExpertProg** [comment]
If anybody runs into this, it might be because you set the default `dtype` to something other than `torch.float32`, and the tensor is initialized with that instead of `float32`. I know it was fixed anybody using `triton<=3.2`/`torch<=2.6` would still need to work around this.
